a/unit-tests/main.lisp

```diff
@@ -32,41 +32,70 @@ If it does, then it is a test package."
         nil
         (string= "LBGE.TEST." (subseq name 0 10)))))
 
-(defun collect-all-packages (&optional selected-package)
-  (let* ((packages (list-all-packages))
-         (lbge-packages (delete-if-not #'test-package-p packages)))
-    (if selected-package
-        (let ((s (string selected-package)))
-          (delete-if-not (lambda (package)
-                           (string= s (subseq (package-name package)
-                                              10
-                                              (+ 10 (length s)))))
-                         lbge-packages))
-        lbge-packages)))
+(defun collect-all-packages ()
+  (delete-if-not #'test-package-p (list-all-packages)))
 
 (defun delete-test-packages ()
-  (let ((all-packages (collect-all-packages)))
-    (mapcar #'delete-package all-packages)))
+  "Find and delete all test packages"
+  (let ((all-test-packages (collect-all-packages)))
+    (mapcar #'delete-package all-test-packages)))
 
-(defun collect-test-packages (&optional selected-package)
-  (filter-disabled-packages (collect-all-packages selected-package)))
+(defun collect-active-tests ()
+  (filter-disabled-packages (collect-all-packages)))
 
 (defun test-package-disabled-p (package)
   (find-symbol "*LBGE-SKIP-TEST*" package))
 
 (defun filter-disabled-packages (package-list)
   (delete-if #'test-package-disabled-p package-list))
 
-(defun run (&key (reporter :spec) selected-package)
+(defun run-selected-test-suite (packages selected-test-suite)
+  (let ((suite (find-package selected-test-suite)))
+    (if suite
+      (rove:run-suite suite)
+      (format t "Could not run test suite: package ~A not found~%" selected-test-suite))))
+
+(defun report-results ()
+  (let ((failed (rove/core/stats:stats-failed rove:*stats*))
+        (passed (rove/core/stats:stats-passed rove:*stats*)))
+
+    (when (> (length passed) 0)
+      (format t "~%-------------~%Passed tests:~%")
+      (loop :for test :across (rove/core/stats:stats-passed rove:*stats*) :do
+        (format t "* ~A~%" (rove/core/result:test-name test))))
+    (when (> (length failed) 0)
+      (format t "~%-------------~%Failed tests:~%")
+      (loop :for test :across failed :do
+        (format t "* ~A~%" (rove/core/result:test-name test))))
+    (when (= (length failed) 0)
```

Comment:
This and the previous `when` can probably be compacted into one `cond` with two cases